### DISCUSSION TITLE
How can i get the advance-width table in hmtx?

### DISCUSSION THREAD

**Discussion by hot-dog** [fonttools/fonttools#2375] [Q&A] [answered]
I have tried font["hmtx"].aWidth and font["hmtx"].advanceWidth, but it raises AttributeError: 'table__h_m_t_x' object has no attribute 'aWidth'.
So is there a API that can give me the advance-width table in hmtx? Or how can i get the advance-width table? Thank you!

**Comment by justvanrossum** [comment, marked as answer]
Here's an example: https://github.com/fonttools/fonttools/wiki/How-do-I...%3F#metrics-how-to-measure-various-things

**Reply by arialcrime** [reply]
Have there been any internal changes that make this snippet not work anymore in recent fonttools versions? Using fonttools version 4.49.0 the CFF specific line throws the following error for me:
`AttributeError: '_TTGlyphCFF' object has no attribute '_glyph'`

**Reply by anthrotype** [reply]
this line?

> bounds = font.getGlyphSet()[glyphname]._glyph.calcBounds(font.getGlyphSet())

hm it shouldn't have tried to access a private attribute (prefixed with "_")... Those aren't guaranteed to stay unchanged.
What exactly are you trying to get? The `width` should be already exposed in the `_TTGlyph` API. Is it the bounding box of a CFF glyph?

i think you can get the charstring object from the CFF directly (what was once the `_glyph` attribute value that's no longer there)
with something like

```
font = TTFont(...)
charstrings = list(font["CFF"].cff.values())[0].CharStrings
glyph = charstrings["A"]
bounds = glyph.calcBounds(charstrings)
```

maybe we could add a method or property to the _TTGlyph object so this works for both glyf and CFF glyphs

**Comment by hot-dog** [comment]
Thank you very much, it really do me a favor!
